File: GestaoEventos API/CarrosAPI2/CarrosAPI2/Models/Utilizador.cs
using System.ComponentModel.DataAnnotations;

namespace GestaoEventosAPI.Models
{
    public class Utilizador
    {
        [Key]
        public int Id_Utilizador { get; set; }

        [Required]
        [StringLength(100)]
        public string Nome { get; set; }

        public string Email { get; set; }

        public string Telemovel { get; set; }

        public DateTime Data_Nascimento { get; set; }

        [Required]
        [StringLength(100)]
        public string passe { get; set; }  // adiciona aqui a password
    }
}

File: gestao_eventos - Flutter/gestao_eventos/lib/Models/utilizador.dart
class Utilizador {
  final int? idUtilizador;
  final String Nome;
  final String Email;
  final String Telemovel;
  final DateTime Data_Nascimento;
  final String passe;

  Utilizador({
    this.idUtilizador,
    required this.Nome,
    required this.Email,
    required this.Telemovel,
    required this.Data_Nascimento,
    required this.passe,
  });

  factory Utilizador.fromJson(Map<String, dynamic> json) {
    return Utilizador(
      idUtilizador: json['id_Utilizador'],
      Nome: json['nome'] ?? '',
      Email: json['email']?.trim() ?? '',
      Telemovel: json['telemovel'] ?? '',
      Data_Nascimento: DateTime.parse(json['data_Nascimento']),
      passe: json['passe'] ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
        if (idUtilizador != null) 'id_Utilizador': idUtilizador,
        'nome': Nome,
        'email': Email,
        'telemovel': Telemovel,
        'data_Nascimento': Data_Nascimento.toIso8601String(),
        'passe': passe,
      };
}
